design/external-references.md

```diff
@@ -0,0 +1,394 @@
+# RFC: External Resource References
+
+## Background
+
+PR [#1436](https://github.com/awslabs/aws-cdk/pull/1436) had recently introduced implicit
+references of resource attributes across stacks within the same CDK app:
+
+
+```ts
+const user = new iam.User(stackWithUser, 'User');
+const group = new iam.Group(stackWithGroup, 'Group');
+
+group.addUser(user);
+```
+
+Will synthesize the following two templates:
+
+```yaml
+Outputs:
+  ExportsOutputRefGroupC77FDACD8CF7DD5B:
+    Value:
+      Ref: GroupC77FDACD
+    Export:
+      Name: ExportsOutputRefGroupC77FDACD8CF7DD5B
+Resources:
+  GroupC77FDACD:
+    Type: AWS::IAM::Group
+```
+
+and:
+
+```yaml
+Resources:
+  User00B015A1:
+    Type: AWS::IAM::User
+    Properties:
+      Groups:
+      - Fn::ImportValue: ExportsOutputRefGroupC77FDACD8CF7DD5B
+```
+
+You can see that the reference automatically resulted in an `Output+Export` and
+an `Fn::ImportValue` to be used in order to share this information across the
+stacks within the same application.
+
+This mechanism practically replaces our "import/export" pattern which looks like this (same use case and same synthesized output):
+
+```ts
+const user = new iam.User(stackWithUser, 'User');
+const group = new iam.Group(stackWithGroup, 'Group');
+
+// creates an `Output` in "stackWithUser" with a unique export name
+// returns an `Fn::ImportValue` that can be used to import this user
+// in any stack in the same account/region
+const importValues = user.export();
+
+// instantiates an `IUser` whose attributes are `Fn::ImportValue` tokens
+// in "stackWithGroup", and it can now be used "naturally" in that stack
+const importedUser = iam.User.import(stackWithGroup, 'ImportedUser', importValues);
+group.addUser(importedUser);
+```
+
+Obvsiouly the world is a better place now. Thanks [@rix0rrr](https://github.com/users/rix0rrr).
+
+## Problem Statement
+
+So, "what seems to be the problem?", you ask. Well, we want more! The current model only supports references that occur between stacks within the same CDK app. What about the use case where I want to reference a resource that was created in a totally different app?
```

Comment:
I usually wrap manually but wanted to play around without wrapping, see how the experience is. Why do you feel the source would read better with manual wrapping?

Reply:
When I started reviewing I felt mildly bothered by the line length. I noted it as a readability issue, but the more I think about it, the more I think the real thing is I felt like it impairs my ability to submit a review comment "as granular" as I'd have liked (since the comment is on the line). Shorter lines mean more "geography-specific" commenting, I guess.

It's not a major thing though, I think by the end of the review I didn't feel so frustrated anyway, so maybe it's just a case of getting used to it?